pkg/resource/deploy/source_eval_test.go

```diff
@@ -194,7 +194,7 @@ func newTestPluginContext(t testing.TB, program deploytest.ProgramFunc) (*plugin
 	statusSink := diagtest.LogSink(t)
 	lang := deploytest.NewLanguageRuntime(program)
 	host := deploytest.NewPluginHost(sink, statusSink, lang)
-	return plugin.NewContext(sink, statusSink, host, nil, "", nil, false, nil)
+	return plugin.NewContext(context.Background(), sink, statusSink, host, nil, "", nil, false, nil)
```

Comment:
```suggestion
	return plugin.NewContext(t.Context(), sink, statusSink, host, nil, "", nil, false, nil)
```

Reply:
We need to wait for pkg to be on go 1.24 to use `t.Context()`.